### PR TITLE
Adds Wrestling via the Championship belt for the Quartermaster.

### PR DESCRIPTION
**Pull request by JohnFulpWillard**
## About The Pull Request

This adds a Fulp-only version of the Wrestling martial art, and gives it to the Championship belt in the Vault. This can only be used by the Quartermaster, and the differences between the Fulp version and the main one (which is still in the game) is the damage, stuns, and sleeps. Most damage for half of what they used to, kick doesn't do damage anymore, stuns are either halved or 1/3 of what they were, and no moves can put you to sleep at all.

## Why It's Good For The Game

Quartermaster can finally join the Warden in having something unique as a semi-head to guard their department and themselves with, since they are a target (Revolutionaries and Traitors with vault objectives, for example) yet don't even get a baton, let alone a flash.

## Changelog
:cl:
add: A new version of the Wrestling Martial Art
balance: Championship belt now gives the Quartermaster the Fulp-only Wrestling martial art
balance: Quartermaster Wrestling's damage has been incredibly reduced, stuns deal almost nothing and putting people to sleep isn't a thing anymore.
/:cl:

### CODE REVIEW

**Comment by ATH1909** [comment]
To be fair, the average joe doesn't learn wrestliing from putting on gloves, either.

You could probably justify letting this belt give the ability to wrestle to anyone by just using nanomachines as an excuse. If you do that, maybe you could add the ability to purchase more belts from cargo so that you can have actual wrestling duels (instead of one guy beating the shit out of another guy because he knows how to wrestle and the other guy doesn't). I mean, I get that a truely great wrestler can get a good match out of a coatrack, but still...

**Comment by JohnFulpWillard** [comment]
> To be fair, the average joe doesn't learn wrestliing from putting on gloves, either.
> 
> You could probably justify letting this belt give the ability to wrestle to anyone by just using nanomachines as an excuse. If you do that, maybe you could add the ability to purchase more belts from cargo so that you can have actual wrestling duels (instead of one guy beating the shit out of another guy because he knows how to wrestle and the other guy doesn't). I mean, I get that a truely great wrestler can get a good match out of a coatrack, but still...

Wrestling matches would be amazing, but an entire fighting style anyone could get is kinda OP, and the reason I don’t want this to be “nanochips” is because you’re wearing a belt, not a pair of advanced gloves

**Comment by ATH1909** [comment]
> > To be fair, the average joe doesn't learn wrestliing from putting on gloves, either.
> > You could probably justify letting this belt give the ability to wrestle to anyone by just using nanomachines as an excuse. If you do that, maybe you could add the ability to purchase more belts from cargo so that you can have actual wrestling duels (instead of one guy beating the shit out of another guy because he knows how to wrestle and the other guy doesn't). I mean, I get that a truely great wrestler can get a good match out of a coatrack, but still...
> 
> Wrestling matches would be amazing, but an entire fighting style anyone could get is kinda OP, and the reason I don’t want this to be “nanochips” is because you’re wearing a belt, not a pair of advanced gloves

If it'd be OP if any crewmember could get it, then it'd probably also be OP in the hands of the QM.

**Comment by ATH1909** [comment]
Okay, I just looked up the exact details of the wrestling martial art style (https://github.com/tgstation/tgstation/blob/master/code/datums/martial/wrestling.dm), and sweet mother of Jesus, whatever you do, do NOT give this... _thing_ to anyone other than a nuke op or a wizard. I'd say it's worth about 25 nuke op TC, and at least 4 wizard points.

This martial art is admin-only for a _reason_- it's less the "duel between wrestlers" martial art that I thought it was, and more of a "one man stylishly dunks on the entirety of the station single-handedly, doesn't even break a sweat while doing so" martial art. The wrestling martial art's entire kit basically revolves around you clicking on someone in melee ONCE (with an empty hand, so the only tell would be the belt you're wearing) to hardstun them for 6-10 seconds (or a headbutt to knock them out for 2 seconds), then performing a variety of stylish moves on them to chainstun them to death. The martial art's currently design basically cannot be reworked to use the stamina system or be appropriate for usage by normal crewmembers or normal antags, as without its hardstuns, it has a hard time setting up its moves (many of which require a grab or your target to stay next to you for a second).

This martial art is NOT designed with being "fair" in mind- it's designed to make a murderbone with it entertaining to watch (which it looks like it does pretty well (if the person using it knows what they're doing), from what I can tell from the code).

**Comment by JohnFulpWillard** [comment]
> 
> 
> Okay, I just looked up the exact details of the wrestling martial art style (https://github.com/tgstation/tgstation/blob/master/code/datums/martial/wrestling.dm), and sweet mother of Jesus, whatever you do, do NOT give this... _thing_ to anyone other than a nuke op or a wizard. I'd say it's worth about 25 nuke op TC, and at least 4 wizard points.
> 
> This martial art is admin-only for a _reason_- it's less the "duel between wrestlers" martial art that I thought it was, and more of a "one man stylishly dunks on the entirety of the station single-handedly, doesn't even break a sweat while doing so" martial art. The wrestling martial art's entire kit basically revolves around you clicking on someone in melee ONCE (with an empty hand, so the only tell would be the belt you're wearing) to hardstun them for 6-10 seconds (or a headbutt to knock them out for 2 seconds), then performing a variety of stylish moves on them to chainstun them to death. The martial art's currently design basically cannot be reworked to use the stamina system or be appropriate for usage by normal crewmembers or normal antags, as without its hardstuns, it has a hard time setting up its moves (many of which require a grab or your target to stay next to you for a second).
> 
> This martial art is NOT designed with being "fair" in mind- it's designed to make a murderbone with it entertaining to watch (which it looks like it does pretty well (if the person using it knows what they're doing), from what I can tell from the code).

I've been messing around with this for hours on a private server and it can be beaten by a single stunbaton, disabler gun, or any other martial art including Krav.

**Comment by AlucardHellsingChapel** [comment]
Or wheelchair t5

**Comment by ATH1909** [comment]
> > Okay, I just looked up the exact details of the wrestling martial art style (https://github.com/tgstation/tgstation/blob/master/code/datums/martial/wrestling.dm), and sweet mother of Jesus, whatever you do, do NOT give this... _thing_ to anyone other than a nuke op or a wizard. I'd say it's worth about 25 nuke op TC, and at least 4 wizard points.
> > This martial art is admin-only for a _reason_- it's less the "duel between wrestlers" martial art that I thought it was, and more of a "one man stylishly dunks on the entirety of the station single-handedly, doesn't even break a sweat while doing so" martial art. The wrestling martial art's entire kit basically revolves around you clicking on someone in melee ONCE (with an empty hand, so the only tell would be the belt you're wearing) to hardstun them for 6-10 seconds (or a headbutt to knock them out for 2 seconds), then performing a variety of stylish moves on them to chainstun them to death. The martial art's currently design basically cannot be reworked to use the stamina system or be appropriate for usage by normal crewmembers or normal antags, as without its hardstuns, it has a hard time setting up its moves (many of which require a grab or your target to stay next to you for a second).
> > This martial art is NOT designed with being "fair" in mind- it's designed to make a murderbone with it entertaining to watch (which it looks like it does pretty well (if the person using it knows what they're doing), from what I can tell from the code).
> 
> I've been messing around with this for hours on a private server and it can be beaten by a single stunbaton, disabler gun, or any other martial art including Krav.

Have you tried headbutting someone while you're knocked down?

**Comment by ATH1909** [comment]
Furthermore, you knew that the person with wrestling was a bad guy in the first place. A QM who cares more about combat effectiveness than actually styling on people with wrestling can just spam the headbutt move to instantly knock you out and keep you knocked out as they kill you with 10-15 brute damage per headbutt.

**Comment by ATH1909** [comment]
Like, you can add this if you really want to, but I'm just trying to warn you that making the QM, a role that's not supposed to validhunt, a better martial artist than the warden (or even the chef while they're in their kitchen) _might_ be a bad idea.

On the bright side, it'll mean that less QMs will go full gun cargo, as they can effortlessly dunk on any antag (bar a nuke op or certain wizard builds) with just their bare hands (and this belt). The belt being locked in the vault doesn't mean anything, by the way, as the QM's cargo door remote can unbolt the vault's bolted door.

**Comment by ATH1909** [comment]
> Or wheelchair t5

Do t5 wheelchairs still crush people who're on scooters or secways? Can they move in 0 g?

**Comment by ATH1909** [comment]
Me:
> The martial art's currently design basically cannot be reworked to use the stamina system or be appropriate for usage by normal crewmembers or normal antags, as without its hardstuns, it has a hard time setting up its moves (many of which require a grab or your target to stay next to you for a second).

John:
> Goes and nerfs the stunlengths, making it impractical to use the wrestling moves instead of just chainstunning someone with the grab and beating them with a blunt object.
> Wrestling is still broken as fuck, but now it's much harder to freestyle with it.

Why do I even bother?

**Comment by ATH1909** [comment]
The solution to the problem wasn't to nerf wrestling, it was to not give wrestling to normal crew members. A martial art (in SS13) based around Bayonetta'ing someone to death _can_ work as a _fun_ concept, but it _can't_ work as a _balanced_ concept. Either you make it impossible (or at least very tight timing-wise) to set up any of its combos, resulting in an impractical mess that isn't useful for anything, or you make it possible to 0 to death someone. There's not really much in-between.

**Comment by JohnFulpWillard** [comment]
> 
> 
> The solution to the problem wasn't to nerf wrestling, it was to not give wrestling to normal crew members. A martial art (in SS13) based around Bayonetta'ing someone to death _can_ work as a _fun_ concept, but it _can't_ work as a _balanced_ concept. Either you make it impossible (or at least very tight timing-wise) to set up any of its combos, or you make it possible to 0 to death someone. There's not really much in-between.

Again, even before I nerfed it I was able to beat it with a single stun baton, disabler, or every other martial art. This only works for the Quartermaster anyways, so they don't have a reason to go cargunia or greytide into other places for weapons.
My main goal with this is just to give the Quartermaster _something_ to defend cargo with from especially the Greytide, who can just shove the average QM down and steal their shit

**Comment by ATH1909** [comment]
> > The solution to the problem wasn't to nerf wrestling, it was to not give wrestling to normal crew members. A martial art (in SS13) based around Bayonetta'ing someone to death _can_ work as a _fun_ concept, but it _can't_ work as a _balanced_ concept. Either you make it impossible (or at least very tight timing-wise) to set up any of its combos, or you make it possible to 0 to death someone. There's not really much in-between.
> 
> Again, even before I nerfed it I was able to beat it with a single stun baton, disabler, or every other martial art. This only works for the Quartermaster anyways, so they don't have a reason to go cargunia or greytide into other places for weapons.

Who were you testing it with? Did that person know how to conterplay against disablers? What if the QM gets culted and grabs a mirror shield?

"let's give the QM a massively overpowered weapon so that he doesn't feel the need to purchase weapons"

What's wrong with gun cargo that isn't wrong with the QM being able to stun anyone in one hit?

File: code/datums/martial/wrestling.dm [hunk 1]
```diff
@@ -354,7 +359,7 @@
 					"<span class='userdanger'>You're roundhouse-kicked by [A]!</span>", "<span class='hear'>You hear a sickening sound of flesh hitting flesh!</span>", COMBAT_MESSAGE_RANGE, A)
 	to_chat(A, "<span class='danger'>You roundhouse-kick [D]!</span>")
 	playsound(A.loc, "swing_hit", 50, TRUE)
-	D.adjustBruteLoss(rand(10,20))
+	D.adjustBruteLoss(rand(2,5))
```
Comment (ATH1909) [inline code/datums/martial/wrestling.dm:362]: B R U H

Why would I EVER use this move? It's worse than just punching someone on harm intent (and flipping/screaming using a hotkey).

Comment (ATH1909) [inline code/datums/martial/wrestling.dm]: damn bro I didn't know that a roundhouse kick does less damage on average than a normal punch

**Comment by ATH1909** [comment]
I hope I'm also getting another point across here: If this PR gets merged, people will unironically start rolling QM to validhunt instead of sec officer, just as some people roll chaplain to heal instead of medical doctor.

**Comment by ATH1909** [comment]
Also, I'm pretty sure that cult soulstones (NOT the chaplain's "old shard") work on people who're unconscious. A QM cultist with quickdraw hotkeys might be able to headbutt someone, then soulstone them before 1 second passes. Maybe that's a different kind of unconsciousness, though, IDK.

**Comment by ATH1909** [comment]
By the way, John, you should join the Coderbus Discord server, since you're a coder now and all.

**Comment by AlucardHellsingChapel** [comment]
Just make it work only in cargo

**Comment by manray0** [comment]
Yeah wrestling is OP and should stay as admin only IMO, makes joining the game as Dave Bautista that much more enjoyable

**Comment by ATH1909** [comment]
You might want to mention the "nerfs" you made to wrestling in the OP and the changelog.

**Comment by JohnFulpWillard** [comment]
I should also mention that the changes I made to the wrestling.dm wasn't me, I copied over the latest version of it from /tg/'s code directly, since they recently changed it

**Comment by ATH1909** [comment]
> I should also mention that the changes I made to the wrestling.dm wasn't me, I copied over the latest version of it from /tg/'s code directly, since they recently changed it

Uh, when? https://github.com/tgstation/tgstation/pull/52055 never got merged.

Oh, do you mean https://github.com/tgstation/tgstation/pull/53221?

**Comment by JohnFulpWillard** [comment]
> 
> 
> > I should also mention that the changes I made to the wrestling.dm wasn't me, I copied over the latest version of it from /tg/'s code directly, since they recently changed it
> 
> Uh, when? [tgstation#52055](https://github.com/tgstation/tgstation/pull/52055) never got merged.
> 
> Oh, do you mean [tgstation#53221](https://github.com/tgstation/tgstation/pull/53221)?

most likely is

**Comment by ATH1909** [comment]
And I see that you've doubled down on trying to make this "balanced" for crew usage.

**Comment by JohnFulpWillard** [comment]
> 
> 
> And I see that you've doubled down on trying to make this "balanced" for crew usage.

I still want this to be Quartermaster-only, but I don't want it to be used to kill whoever they want, only to kick greytiders out of cargo with style rather than guns

**Comment by ATH1909** [comment]
> > And I see that you've doubled down on trying to make this "balanced" for crew usage.
> 
> I still want this to be Quartermaster-only, but I don't want it to be used to kill whoever they want, only to kick greytiders out of cargo with style rather than guns

I still don't get why this has to be QM-only. The krav maga gloves are intended to be used by the warden, but they can still be used by other people. Just say it gives you the wrestling martial art using "nanomachines" (like the krav maga gloves) that tap into your spine or spread throughout your bloodstream or something. The QM isn't the only person who might want to do a wrestling gimmick, after all...

**Comment by ErwinSnowden** [comment]
ATH1909 is over-exaggerating the usefulness of wrestling, as if you get stunbatonned you're OUT FOR THE COUNT, wrestling needs combat stims to be useful in any combat encounter that isn't 1 on 1.

I'd honestly prefer if the wrestling belt was a traitor item for some majority of your telecrystals though. (Also wrestling has an instant aggro grab which is a poor choice and people will play QM just to rush the belt, wont solve tiding, will instead make one crew a gamer machine)

**Comment by JohnFulpWillard** [comment]
> 
> 
> ATH1909 is over-exaggerating the usefulness of wrestling, as if you get stunbatonned you're OUT FOR THE COUNT, wrestling needs combat stims to be useful in any combat encounter that isn't 1 on 1.
> 
> I'd honestly prefer if the wrestling belt was a traitor item for some majority of your telecrystals though. (Also wrestling has an instant aggro grab which is a poor choice and people will play QM just to rush the belt, wont solve tiding, will instead make one crew a gamer machine)

there's only a chance to aggro grab, and I also nerfed the stun you get from it to almost nothing (and I might as well just remove it entirely), and aggro grabs are easy to break out of. Also, I don't often see people play Warden just for the Krav Maga gloves. Maybe in the first few weeks like when anything is added, people will want to mess with it, but overtime it should die down

**Comment by BlueMemesauce** [comment]
I think wrestling lets you chop peoples limbs off

**Comment by JohnFulpWillard** [comment]
> 
> 
> I think wrestling lets you chop peoples limbs off

Like any weapon, If they're not wearing a jumpsuit, then yeah you have a chance at ripping their limbs off if they're somewhat damaged already.

**Comment by ATH1909** [comment]
image
It's like I'm talking to a brick wall.

**Comment by JohnFulpWillard** [comment]
> 
> 
> image
> It's like I'm talking to a brick wall.

Hardstunning grabs is broken when your strongest moves require them being in a grab, they should be getting them in a cinch in otherways other than one click win.

**Comment by ATH1909** [comment]
> ATH1909 is over-exaggerating the usefulness of wrestling, as if you get stunbatonned you're OUT FOR THE COUNT, wrestling needs combat stims to be useful in any combat encounter that isn't 1 on 1.
> 
> I'd honestly prefer if the wrestling belt was a traitor item for some majority of your telecrystals though. (Also wrestling has an instant aggro grab which is a poor choice and people will play QM just to rush the belt, wont solve tiding, will instead make one crew a gamer machine)

Just chainstrike or chainheadbutt them, bro. Being knocked down doesn't count as being incapacitated for selecting moves, right?

Also, it wasn't an instant aggressive grab (in fact, people who know the wrestling martial art are physically incapable of normally starting an aggressive grab, IIRC), it was a hardstun.

**Comment by E231kei500bandai** [comment]
Why are we giving this to the Quartermaster anyway. Even if it is nerfed alot its still a weapon your giving to the qm for no reason. You're not allowed to defend cargo on fulp you have to call security it make no sense to add this. Plus the qm can just buy guns if he had a legitimate reason to want a weapon

**Comment by JohnFulpWillard** [comment]
> Just chainstrike or chainheadbutt them, bro. Being knocked down doesn't count as being incapacitated for selecting moves, right?

from my testing, yes it does
> Also, it wasn't an instant aggressive grab (in fact, people who know the wrestling martial art are physically incapable of normally starting an aggressive grab, IIRC), it was a hardstun.

yeah that's my bad, it had a chance to grab you, and grabs would hardstun (which I removed)

> Why are we giving this to the Quartermaster anyway. Even if it is nerfed alot its still a weapon your giving to the qm for no reason. You're not allowed to defend cargo on fulp you have to call security it make no sense to add this. Plus the qm can just buy guns if he had a legitimate reason to want a weapon

So they have a fun thing to defend their cargo on 1v1's against greytider/revs without the need to call Security/break server rules

**Comment by ErwinSnowden** [comment]
> > Just chainstrike or chainheadbutt them, bro. Being knocked down doesn't count as being incapacitated for selecting moves, right?
> 
> from my testing, yes it does
> 
> > Also, it wasn't an instant aggressive grab (in fact, people who know the wrestling martial art are physically incapable of normally starting an aggressive grab, IIRC), it was a hardstun.
> 
> yeah that's my bad, it had a chance to grab you, and grabs would hardstun (which I removed)
> 
> > Why are we giving this to the Quartermaster anyway. Even if it is nerfed alot its still a weapon your giving to the qm for no reason. You're not allowed to defend cargo on fulp you have to call security it make no sense to add this. Plus the qm can just buy guns if he had a legitimate reason to want a weapon
> 
> So they have a fun thing to defend their cargo on 1v1's against greytider/revs without the need to call Security/break server rules



> > Just chainstrike or chainheadbutt them, bro. Being knocked down doesn't count as being incapacitated for selecting moves, right?
> 
> from my testing, yes it does
> 
> > Also, it wasn't an instant aggressive grab (in fact, people who know the wrestling martial art are physically incapable of normally starting an aggressive grab, IIRC), it was a hardstun.
> 
> yeah that's my bad, it had a chance to grab you, and grabs would hardstun (which I removed)
> 
> > Why are we giving this to the Quartermaster anyway. Even if it is nerfed alot its still a weapon your giving to the qm for no reason. You're not allowed to defend cargo on fulp you have to call security it make no sense to add this. Plus the qm can just buy guns if he had a legitimate reason to want a weapon
> 
> So they have a fun thing to defend their cargo on 1v1's against greytider/revs without the need to call Security/break server rules

Sorry sir, read the sign. No fun allowed.

**Pull request closed without merging**